File: src/main/java/ec/ferchoc/java/random/BinarySearch.java
package ec.ferchoc.java.random;

public class BinarySearch {

    /**
     * Implement binary search to find an item.
     * @param sortedList A sorted list.
     * @param item The item to find.
     * @return null if item not found or the position of element.
     */
    public Integer searchItem(int [] sortedList, int item) {

        var low = 0;
        var high = sortedList.length - 1;

        while (low <= high) {
            var mid = (high + low) / 2; ////lowerBound + ( upperBound - lowerBound ) / 2
            var guest = sortedList[mid];
            if (guest == item) {
                return mid;
            } else if (guest > item) {
                high = mid - 1;
            } else {
                low = mid + 1;
            }
        }

        return null;

    }

}

File: src/main/java/ec/ferchoc/kotlin/random/BinarySearch.kt
package ec.ferchoc.kotlin.random

/**
 * Implements a simple binary search.
 */
class BinarySearch {

    fun searchItem(sortedList: IntArray, target: Int) : Int {
        var low = 0
        var high = sortedList.size - 1
        while (low <= high) {
            val middle = low + (high - low)/2
            val guest = sortedList[middle]
            when {
                guest == target -> return middle
                guest > target -> high = middle - 1
                else -> {
                    low = middle + 1
                }
            }
        }
        return -1;
    }
}